src/doc/reference.md

```diff
@@ -653,7 +657,7 @@ There are several kinds of item:
 * [`use` declarations](#use-declarations)
 * [modules](#modules)
 * [functions](#functions)
-* [type definitions](#type-definitions)
+* [type definitions](grammar.html#type-definitions)
```

Comment:
I think this should just link to the type aliases section in the same chapter 6.1. I touched this in my PR too (added to this particular list, missing constants).